src/pubsub.rs

```diff
@@ -0,0 +1,635 @@
+//! The `pubsub` module implements a threaded subscription service on client RPC request
+
+use bank::Bank;
+use bs58;
+use jsonrpc_core::futures::Future;
+use jsonrpc_core::*;
+use jsonrpc_macros::pubsub;
+use jsonrpc_pubsub::{PubSubHandler, PubSubMetadata, Session, SubscriptionId};
+use jsonrpc_ws_server::ws;
+use jsonrpc_ws_server::{RequestContext, Sender, ServerBuilder};
+use rpc::{JsonRpcRequestProcessor, RpcSignatureStatus};
+use signature::{Keypair, KeypairUtil, Signature};
+use solana_program_interface::account::Account;
+use solana_program_interface::pubkey::Pubkey;
+use std::collections::HashMap;
+use std::mem;
+use std::net::SocketAddr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{atomic, Arc, Mutex, RwLock};
+use std::thread::{sleep, Builder, JoinHandle};
+use std::time::Duration;
+
+pub enum ClientState {
+    Uninitialized,
+    Init(Sender),
+}
+
+#[derive(Serialize)]
+pub struct SubscriptionResponse {
+    pub port: u16,
+    pub path: String,
+}
+
+pub struct PubSubService {
+    _thread_hdl: JoinHandle<()>,
+}
+
+impl PubSubService {
+    pub fn new(
+        bank: &Arc<Bank>,
+        pubsub_addr: SocketAddr,
+        path: Pubkey,
+        exit: Arc<AtomicBool>,
+    ) -> Self {
+        let request_processor = JsonRpcRequestProcessor::new(bank.clone());
+        let status = Arc::new(Mutex::new(ClientState::Uninitialized));
+        let client_status = status.clone();
+        let server_bank = bank.clone();
+        let _thread_hdl = Builder::new()
+            .name("solana-pubsub".to_string())
+            .spawn(move || {
+                let mut io = PubSubHandler::default();
+            	let rpc = RpcSolPubSubImpl::default();
```

Comment:
`cargo fmt` allowed this?

Reply:
That's odd. Those indentations look correct and aligned for me locally, and git says everything is up to date with the remote.
That said, `cargo fmt` seems to be ignoring the `Builder` expression when I deliberately screw it up.
I'll try formatting it from scratch and push with some other changes and see if that makes a difference.